app/src/main/java/com/github/polypoly/app/network/RemoteDB.kt

```diff
@@ -51,6 +51,14 @@ open class RemoteDB(
         return valuePromise
     }
 
+    override fun getSnapshot(key: String): CompletableFuture<DataSnapshot> {
+        return getValueAndThen(key, { data, valuePromise ->
+            valuePromise.complete(data)
+        }, { valuePromise ->
+            valuePromise.completeExceptionally(IllegalAccessException("No value found for key <$key>"))
```

Comment:
same remark as in another PR, this exception is meant for illegal reflection accesses to class members, NoSuchElementException would probable be a better fit, or maybe create your own exception type

Reply:
I agree with that, this is not my code but i'll mention @Jharaxus on that